src/core/labeling/qgspallabeling.cpp

```diff
@@ -563,6 +563,11 @@ void QgsPalLayerSettings::stopRender( QgsRenderContext &context )
   mRenderStarted = false;
 }
 
+bool QgsPalLayerSettings::containsAdvancedEffects() const
+{
+  return mFormat.containsAdvancedEffects() || mCallout->blendMode() != QPainter::CompositionMode_SourceOver;
```

Comment:
Would be cleaner to make a new containsAdvancedEffects virtual method in QgsCallout